src/workerd/jsg/resource.h

```diff
@@ -191,6 +192,89 @@ template <typename TypeWrapper,
     typename Indexes>
 struct MethodCallback;
 
+// Helper to determine if a type is compatible with V8 fast API
+// TODO(soon): Add support for v8::Local<v8::Value>
+template <typename T>
+struct IsFastApiCompatibleType {
+  static constexpr bool value = std::is_same_v<T, void> || std::is_same_v<T, bool> ||
+      std::is_same_v<T, int32_t> || std::is_same_v<T, uint32_t> || std::is_same_v<T, float> ||
+      std::is_same_v<T, double> ||
+      std::is_pointer_v<T>;  // Allow pointers, which can be embedder types
+};
+
+// Helper to determine if a method can be used with V8 fast API
+template <typename Ret, typename... Args>
+struct IsFastApiCompatible {
+  static constexpr bool value =
+      IsFastApiCompatibleType<Ret>::value && (IsFastApiCompatibleType<Args>::value && ...);
+};
+
+// Helper to determine if a method pointer type is compatible with Fast API
+template <typename Method>
+struct IsFastMethodCompatible {
+  static constexpr bool value = false;
+};
+
+// Specialization for non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...)> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
+
+// Specialization for const non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...) const> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
+
+// Implements V8 Fast API callback for a method. This class handles both the fast and slow paths.
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename Method,
+    Method method,
+    typename Indexes>
+struct FastMethodCallback;
+
+// Specialization for instance methods compatible with Fast API
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename U,
+    typename Ret,
+    typename... Args,
+    Ret (U::*method)(Args...),
+    size_t... indexes>
+struct FastMethodCallback<TypeWrapper,
+    methodName,
+    isContext,
+    T,
+    Ret (U::*)(Args...),
+    method,
+    kj::_::Indexes<indexes...>> {
+
+  static void slowCallback(const v8::FunctionCallbackInfo<v8::Value>& args) {
+    MethodCallback<TypeWrapper, methodName, isContext, T, Ret (U::*)(Args...), method,
+        kj::_::Indexes<indexes...>>::callback(args);
+  }
+
+  static Ret fastCallback(
+      v8::Local<v8::Object> receiver, Args... fastArgs, v8::FastApiCallbackOptions& options) {
+    v8::Local<v8::Context> context = options.isolate->GetCurrentContext();
+
+    auto& self = extractInternalPointer<T, isContext>(context, receiver);
+
+    return (self.*method)(fastArgs...);
```

Comment:
This will need to catch and handle exceptions (e.g. `kj::Exception`, `jsg::JsExceptionThrown`) that are thrown by the method. Look for `liftKj` within `workerd/jsg` for details.

Reply:
Yes, I'll add liftKj. That's a good recommendation!